### ISSUE TITLE
[Bug] NSRangeException thrown in grouped CollectionView on iOS 10

### ISSUE THREAD

**Issue by velocitysystems** [xamarin/Xamarin.Forms#9058]
### Description
An `NSRangeException` is thrown with a grouped `CollectionView` on iOS 10.
This may possibly be related to #7393.

### Steps to Reproduce
1. Open the ControlGallery app on an iOS 10 device.
2. Navigate to Collection View Galleries >> Grouping Galleries 
3. Tap on the "Grouping, some empty groups" sample.
4. The app will crash.

### Expected Behavior
The gallery page should not crash.

### Actual Behavior
The gallery page crashes with the following exception:
```
Unhandled managed exception: Objective-C exception thrown.  
Reason: *** -[__NSArrayM objectAtIndex:]: index 0 beyond bounds for empty array
```

### Basic Information

- Version with issue: 4.4.0
- Last known good version: N/A
- IDE: Microsoft Visual Studio 2019 Enterprise
- Platform Target Frameworks: 
  - iOS: 13.6.0.12
- Affected Devices: iPad iOS 10.3.3

**Comment by samhouts** [comment]
similar to #9065

**Comment by apwillies** [comment]
We are interested in this fix too. Thanks guys!

**Comment by samhouts** [comment]
@velocitysystems Can you still reproduce this with the latest 4.4.0 branch? I tried with an iPad Air 10.3 sim, and it did not crash.

**Comment by velocitysystems** [comment]
@samhouts Apologies for the delay, have been tied up with a release cycle. Will test this and let you know tomorrow. Thanks!

**Comment by velocitysystems** [comment]
@samhouts Yes, I can confirm this appears to be fixed with the latest 4.4.0 branch.
I don't have an iOS 10 device to test on right now, but can verify it does not crash in the 10.3 simulator (as you mention).

I will mark this as closed and verify #9065 separately. Thanks!
